File: alchemist-loading/src/main/java/it/unibo/alchemist/boundary/variables/Flag.java
/*
 * Copyright (C) 2010-2023, Danilo Pianini and contributors
 * listed, for each module, in the respective subproject's build.gradle.kts file.
 *
 * This file is part of Alchemist, and is distributed under the terms of the
 * GNU General Public License, with a linking exception,
 * as described in the file LICENSE in the Alchemist distribution's top directory.
 */
package it.unibo.alchemist.boundary.variables;

import java.util.stream.Stream;

/**
 * This variable is a flag. Being booleans not a valid data type in charts, this
 * variable just outputs 0 and 1. This is equivalent to a {@link LinearVariable}
 * with two samples ranging from 0 to 1.
 */
public final class Flag extends PrintableVariable<Boolean> {

    private static final long serialVersionUID = 1L;
    private final boolean defVal;

    /**
     * @param def
     *            the default value
     */
    public Flag(final boolean def) {
        this.defVal = def;
    }

    @Override
    public Boolean getDefault() {
        return defVal;
    }

    @Override
    public Stream<Boolean> stream() {
        return Stream.of(true, false);
    }

}

File: alchemist-loading/src/main/kotlin/it/unibo/alchemist/boundary/variables/Flag.kt
/*
 * Copyright (C) 2010-2026, Danilo Pianini and contributors
 * listed, for each module, in the respective subproject's build.gradle.kts file.
 *
 * This file is part of Alchemist, and is distributed under the terms of the
 * GNU General Public License, with a linking exception,
 * as described in the file LICENSE in the Alchemist distribution's top directory.
 */
package it.unibo.alchemist.boundary.variables

import java.io.Serial
import java.util.stream.Stream

/**
 * A boolean flag variable. Charts typically accept numeric values, but the
 * semantic value of this variable is boolean; use `true` and `false` as the
 * two possible outputs. The default value is provided at construction.
 *
 * This is equivalent in behavior to a [LinearVariable] with two samples
 * between 0 and 1, but keeps the boolean semantics.
 *
 * @param default the default boolean value
 */
data class Flag(override val default: Boolean) : AbstractPrintableVariable<Boolean>() {

    override fun stream(): Stream<Boolean> = Stream.of(true, false)

    private companion object {
        @Serial
        private const val serialVersionUID = 1L
    }
}
